pkg/urbit/vere/io/unix.c

```diff
@@ -1176,21 +1176,21 @@ _unix_initial_update_file(c3_c* pax_c, c3_c* bas_c)
   len_ws = buf_u.st_size;
   dat_y = c3_malloc(len_ws);
 
-  red_ws = read(fid_i, dat_y, len_ws);
+  ssize_t red_i = c3_read(fid_i, dat_y, len_ws);
 
   if ( close(fid_i) < 0 ) {
     u3l_log("error closing initial file %s: %s\r\n",
             pax_c, strerror(errno));
   }
 
-  if ( len_ws != red_ws ) {
-    if ( red_ws < 0 ) {
+  if ( red_i != len_ws ) {
+    if ( red_i < 0 ) {
       u3l_log("error reading initial file %s: %s\r\n",
-              pax_c, strerror(errno));
+              pax_c, strerror(-red_i));
     }
     else {
-      u3l_log("wrong # of bytes read in initial file %s: %d %d\r\n",
-              pax_c, len_ws, red_ws);
+      u3l_log("wrong # of bytes read in initial file %s: %d %lu\r\n",
```

Comment:
I think all these partial read printfs (here and in `events.c`) should use the `%zu` format specifier.